File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"
unsigned int sum_of_squares(unsigned int number){
    unsigned int want = 1;
    for(unsigned int i = number; i > 1; i--){
        want += i * i;
    }
    return want;
}
unsigned int square_of_sum(unsigned int number){
    unsigned int want = 1;
    for(unsigned int i = number; i > 1; i--){
        want += i;
    }
    return want * want;
}
unsigned int difference_of_squares(unsigned int number){
    return square_of_sum(number) - sum_of_squares(number);
}

File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    net = 0
    for num in range(1, number + 1):
        net += num
    return net**2


def sum_of_squares(number):
    net = 0
    for num in range(1, number + 1):
        net += num**2
    return net


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
